loglevels-logback/src/test/scala/akka/management/loglevels/logback/LogLevelRoutesSpec.scala

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2017-2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.management.loglevels.logback
+
+import akka.actor.ExtendedActorSystem
+import akka.http.javadsl.server.MalformedQueryParamRejection
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.model.Uri
+import akka.http.scaladsl.testkit.ScalatestRouteTest
+import akka.management.scaladsl.ManagementRouteProviderSettings
+import org.scalatest.Matchers
+import org.scalatest.WordSpec
+import org.slf4j.LoggerFactory
+import akka.event.{ Logging => ClassicLogging }
+
+
+class LogLevelRoutesSpec extends WordSpec with Matchers with ScalatestRouteTest {
+
+
+  override def testConfigSource: String =
+    """
+      akka.loglevel = INFO
+      """
+
+  val routes = LogLevelRoutes
+    .createExtension(system.asInstanceOf[ExtendedActorSystem])
+    .routes(ManagementRouteProviderSettings(Uri("https://example.com"), readOnly = false))
+
+  "The logback log level routes" must {
+
+    "show log level of a Logger" in {
+      Get("/loglevel?logger=LogLevelRoutesSpec") ~> routes ~> check {
+        responseAs[String]
+      }
+    }
+
+    "change log level of a Logger" in {
+      Put("/loglevel?logger=LogLevelRoutesSpec&level=DEBUG") ~> routes ~> check {
+        response.status should ===(StatusCodes.OK)
+        LoggerFactory.getLogger("LogLevelRoutesSpec").isDebugEnabled should ===(true)
+      }
+    }
+
+    "fail for unknown log level" in {
+      Put("/loglevel?logger=LogLevelRoutesSpec&level=MONKEY") ~> routes ~> check {
+        rejection shouldBe an[MalformedQueryParamRejection]
+      }
+    }
+
+    "not change loglevel if read only" in {
+      val readOnlyRoutes = LogLevelRoutes
+        .createExtension(system.asInstanceOf[ExtendedActorSystem])
+        .routes(ManagementRouteProviderSettings(Uri("https://example.com"), readOnly = true))
+      Put("/loglevel?logger=LogLevelRoutesSpec&level=DEBUG") ~> readOnlyRoutes ~> check {
+        response.status should ===(StatusCodes.Forbidden)
+      }
+    }
+
+    "allow inspecting classic Akka loglevel" in {
+      Get("/akka/classic/loglevel") ~> routes ~> check {
```

Comment:
if the ordinary (logback) is `/loglevel` this could be `/loglevel/akka` ?

Reply:
I thought it was good to put all things Akka under `akka` (did so with the enhancement ones as well) since any number of route providers could add their own routes. The other one `loglevel` is JVM-global though and not strictly related to Akka so that's why I gave that a toplevel path.

Not that I feel very strongly about it...

Reply:
true that is good to use somewhat unique paths. I was thinking that it was a sub-feature of `/loglevel`

The existing cluster management is under `/cluster/members/` so we wouldn't place everything Akka under /akka anyway.

Reply:
Good point, I had forgotten we already had some stuff in here, I think it should be leaf-paths though, so wdyt about `/loglevel/logback` and `/loglevel/akka` in that case?

Reply:
Pushed a change of paths to that last suggestion

Reply:
good, thanks